Makefile

```diff
@@ -247,30 +328,70 @@ docker-build-%:
+	docker manifest push --purge $(KUBEADM_BOOTSTRAP_CONTROLLER_IMG):$(TAG)
+	MANIFEST_IMG=$(KUBEADM_BOOTSTRAP_CONTROLLER_IMG) MANIFEST_TAG=$(TAG) $(MAKE) set-kubeadm-bootstrap-manifest-image
+	$(MAKE) set-kubeadm-bootstrap-manifest-pull-policy
+
+.PHONY: docker-push-kubeadm-control-plane-manifest
+docker-push-kubeadm-control-plane-manifest: ## Push the fat manifest docker image for the kubeadm control plane image.
+	## Minimum docker version 18.06.0 is required for creating and pushing manifest images.
+	docker manifest create --amend $(KUBEADM_CONTROL_PLANE_CONTROLLER_IMG):$(TAG) $(shell echo $(ALL_ARCH) | sed -e "s~[^ ]*~$(KUBEADM_CONTROL_PLANE_CONTROLLER_IMG)\-&:$(TAG)~g")
+	@for arch in $(ALL_ARCH); do docker manifest annotate --arch $${arch} ${KUBEADM_CONTROL_PLANE_CONTROLLER_IMG}:${TAG} ${KUBEADM_CONTROL_PLANE_CONTROLLER_IMG}-$${arch}:${TAG}; done
+	docker manifest push --purge $(KUBEADM_CONTROL_PLANE_CONTROLLER_IMG):$(TAG)
+	MANIFEST_IMG=$(KUBEADM_CONTROL_PLANE_CONTROLLER_IMG) MANIFEST_TAG=$(TAG) $(MAKE) set-kubeadm-control-plane-manifest-image
+	$(MAKE) set-kubeadm-control-plane-manifest-pull-policy
+
+.PHONY: set-core-manifest-image
+set-core-manifest-image:
 	$(info Updating kustomize image patch file for manager resource)
 	sed -i'' -e 's@image: .*@image: '"${MANIFEST_IMG}:$(MANIFEST_TAG)"'@' ./config/core/manager_image_patch.yaml
 
-.PHONY: set-manifest-pull-policy
-set-manifest-pull-policy:
+.PHONY: set-core-manifest-pull-policy
+set-core-manifest-pull-policy:
 	$(info Updating kustomize pull policy file for manager resource)
 	sed -i'' -e 's@imagePullPolicy: .*@imagePullPolicy: '"$(PULL_POLICY)"'@' ./config/core/manager_pull_policy.yaml
 
+.PHONY: set-kubeadm-bootstrap-manifest-image
+set-kubeadm-bootstrap-manifest-image:
+	$(info Updating kustomize image patch file for manager resource)
+	sed -i'' -e 's@image: .*@image: '"${MANIFEST_IMG}:$(MANIFEST_TAG)"'@' ./bootstrap/kubeadm/config/default/manager_image_patch.yaml
+
+.PHONY: set-kubeadm-bootstrap-manifest-pull-policy
+set-kubeadm-bootstrap-manifest-pull-policy:
+	$(info Updating kustomize pull policy file for manager resource)
+	sed -i'' -e 's@imagePullPolicy: .*@imagePullPolicy: '"$(PULL_POLICY)"'@' ./bootstrap/kubeadm/config/default/manager_pull_policy.yaml
+
+.PHONY: set-kubeadm-control-plane-manifest-image
+set-kubeadm-control-plane-manifest-image:
+	$(info Updating kustomize image patch file for manager resource)
+	sed -i'' -e 's@image: .*@image: '"${MANIFEST_IMG}:$(MANIFEST_TAG)"'@' ./controlplane/kubeadm/config/default/manager_image_patch.yaml
+
+.PHONY: set-kubeadm-control-plane-manifest-pull-policy
+set-kubeadm-control-plane-manifest-pull-policy:
+	$(info Updating kustomize pull policy file for manager resource)
+	sed -i'' -e 's@imagePullPolicy: .*@imagePullPolicy: '"$(PULL_POLICY)"'@' ./controlplane/kubeadm/config/default/manager_pull_policy.yaml
```

Comment:
These could be generalized a little

```Makefile
.PHONY: set-manifest-pull-policy
set-manifest-pull-policy:
	$(info Updating kustomize pull policy file for manager resources)
	sed -i'' -e 's@imagePullPolicy: .*@imagePullPolicy: '"$(PULL_POLICY)"'@' $(TARGET_RESOURCE)
```

the calls then would be something like:

```Makefile
	$(MAKE) set-manifest-pull-policy PULL_POLICY=IfNotPresent MANIFEST=[path here]
```

Reply:
sure

Reply:
i went ahead and refactored the set-manifest-image in the same way